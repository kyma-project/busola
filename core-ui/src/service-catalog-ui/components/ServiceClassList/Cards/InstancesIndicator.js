import React from 'react';
import classnames from 'classnames';
import { Tooltip } from 'react-shared';

import { isStringValueEqualToTrue } from 'helpers';
import './InstancesIndicator.scss';

const tooltipDescription = {
  provisionOnlyOnce:
    'You can provision this Service Class only once in a given Namespace.',
  provisionOnlyOnceActive:
    'You can provision this Service Class only once in a given Namespace. It is already provisioned in this Namespace.',
  instancesTooltipInfo: 'This Service Class is provisioned',
  instancesTooltipSingle: 'time.',
  instancesTooltipPlural: 'times.',
};

const CardIndicatorGeneral = ({ active, provisionOnce, children }) => {
  const className = classnames('instances-indicator--general', {
    active,
    'provision-once': provisionOnce,
  });
  return <div className={className}>{children}</div>;
};

export function InstancesIndicator({ numberOfInstances, labels }) {
  const isProvisionedOnlyOnce =
    labels &&
    labels.provisionOnlyOnce &&
    isStringValueEqualToTrue(labels.provisionOnlyOnce);
  return (
    <div className="instances-indicator">
      {isProvisionedOnlyOnce && (
        <Tooltip
          content={
            numberOfInstances > 0
              ? tooltipDescription.provisionOnlyOnceActive
              : tooltipDescription.provisionOnlyOnce
          }
        >
          <CardIndicatorGeneral
            data-e2e-id="card-indicator"
            provisionOnce
            active={numberOfInstances > 0}
          >
            1
          </CardIndicatorGeneral>
        </Tooltip>
      )}
      {!isProvisionedOnlyOnce && numberOfInstances > 0 && (
        <Tooltip
          content={`${
            tooltipDescription.instancesTooltipInfo
          } ${numberOfInstances} ${
            numberOfInstances > 1
              ? tooltipDescription.instancesTooltipPlural
              : tooltipDescription.instancesTooltipSingle
          }`}
        >
          <CardIndicatorGeneral
            data-e2e-id={`instances-provisioned`}
            active={numberOfInstances > 0}
          >
            {numberOfInstances}
          </CardIndicatorGeneral>
        </Tooltip>
      )}
    </div>
  );
}
