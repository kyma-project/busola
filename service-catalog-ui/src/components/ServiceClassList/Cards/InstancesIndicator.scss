.instances-indicator {
    margin: 0;
    margin-left: 10px;
    align-self: flex-start;
    font-family: '72';
    position: relative;

    &--general {
        padding: 2px 6px;
        border-radius: 2px;
        font-size: 12px;
        background-color: #eeeeef;
        color: #515559;

        &.active {
            background-color: #0a6ed1;
            color: #ffffff;
        }

        &.provision-once {
            padding: 0;
            width: 28px;
            height: 28px;
            line-height: 28px;
            font-size: 10px;
            text-align: center;
            border-radius: 4px;

            &:before {
                position: absolute;
                top: 55%;
                left: 50%;
                font-size:16px;
                content: '\E010';
                font-family: SAP-icons;
                font-style: normal;
                font-weight: 400;
                text-align: center;
                display: inline-block;
                text-transform: none;
                speak: none;
                transform: scaleY(-1) translate(-50%, 50%);
                filter: flipv;
            }
        }
    }
}