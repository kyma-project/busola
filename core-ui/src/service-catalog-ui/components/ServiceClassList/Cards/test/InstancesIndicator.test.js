import { render } from 'enzyme';
import React from 'react';
import { InstancesIndicator } from '../InstancesIndicator';

const consoleWarn = jest.spyOn(global.console, 'warn').mockImplementation();
afterAll(() => {
  consoleWarn.mockReset();
});

describe('InstancesIndicator', () => {
  it('Renders without instances and provision only once', () => {
    const component = render(
      <InstancesIndicator
        numberOfInstances={0}
        labels={{ provisionOnlyOnce: 'true', local: 'true' }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with instances and provision only once', () => {
    const component = render(
      <InstancesIndicator
        numberOfInstances={1}
        labels={{ provisionOnlyOnce: 'true', local: 'true' }}
      />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders without instances and no provision only once', () => {
    const component = render(
      <InstancesIndicator numberOfInstances={0} labels={{ local: 'true' }} />,
    );
    expect(component).toMatchSnapshot();
  });
  it('Renders with instances and no provision only once', () => {
    const component = render(
      <InstancesIndicator numberOfInstances={3} labels={{ local: 'true' }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
