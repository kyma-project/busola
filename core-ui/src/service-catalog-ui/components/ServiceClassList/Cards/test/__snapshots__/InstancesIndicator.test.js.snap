// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstancesIndicator Renders with instances and no provision only once 1`] = `
<div
  class="instances-indicator"
>
  <div
    class=""
    style="display:inline"
  >
    <div
      class="instances-indicator--general active"
    >
      3
    </div>
  </div>
</div>
`;

exports[`InstancesIndicator Renders with instances and provision only once 1`] = `
<div
  class="instances-indicator"
>
  <div
    class=""
    style="display:inline"
  >
    <div
      class="instances-indicator--general active provision-once"
    >
      1
    </div>
  </div>
</div>
`;

exports[`InstancesIndicator Renders without instances and no provision only once 1`] = `
<div
  class="instances-indicator"
/>
`;

exports[`InstancesIndicator Renders without instances and provision only once 1`] = `
<div
  class="instances-indicator"
>
  <div
    class=""
    style="display:inline"
  >
    <div
      class="instances-indicator--general provision-once"
    >
      1
    </div>
  </div>
</div>
`;
